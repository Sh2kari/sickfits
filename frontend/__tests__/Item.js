import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: '1',
  title: 'cool item',
  price: 5000,
  description: 'this item is really cool',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg'
};

describe('<Item />', () => {
  it('renders', () => {
    shallow(<ItemComponent item={fakeItem} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const image = wrapper.find('img');

  //   expect(image.props().src).toBe(fakeItem.image);
  //   expect(image.props().alt).toBe(fakeItem.title);
  // });

  // it('renders the price tag and title properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);

  //   expect(
  //     wrapper
  //       .find('PriceTag')
  //       .children()
  //       .text()
  //   ).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });

  // it('renders out buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');

  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link').exists()).toBe(true);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  // });
});
