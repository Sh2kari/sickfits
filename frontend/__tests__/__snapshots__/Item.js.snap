// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "1",
          },
        }
      }
    >
      <a>
        cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    this item is really cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "1",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="1"
    />
    <DeleteItem
      id="1"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
